src/modals/EditModal.tsx

```diff
@@ -0,0 +1,129 @@
+import { useContext, useRef } from "react";
+import {
+  IonButton,
+  IonModal,
+  IonDatetime,
+  IonContent,
+  IonItem,
+} from "@ionic/react";
+import { useTranslation } from "react-i18next";
+import { format } from "date-fns";
+import "./EditModal.css";
+
+import { CyclesContext } from "../state/Context";
+import { useAverageLengthOfCycle } from "../state/CycleInformationHooks";
+import { getNewCyclesHistory } from "../state/CalculationLogics";
+
+interface PropsEditModal {
+  isOpen: boolean;
+  setIsOpen: (newIsOpen: boolean) => void;
+}
+
+const EditModal = (props: PropsEditModal) => {
+  const datetimeRef = useRef<null | HTMLIonDatetimeElement>(null);
+  const { t } = useTranslation();
+
+  const { cycles, updateCycles } = useContext(CyclesContext);
+  const averLengthOfCycle = useAverageLengthOfCycle();
+
+  const nowDate = new Date();
+  nowDate.setHours(0, 0, 0, 0);
+
+  const isActiveDates = (dateString: string) => {
+    if (cycles.length === 0) {
+      return true;
+    }
+    const date = new Date(dateString);
+    date.setHours(0, 0, 0, 0);
+
+    const lastCycleFinish: Date = new Date(cycles[0].startDate);
+    lastCycleFinish.setDate(lastCycleFinish.getDate() + averLengthOfCycle);
+    lastCycleFinish.setHours(0, 0, 0, 0);
+
+    return (
+      date.getTime() < lastCycleFinish.getTime() ||
+      date.getTime() <= nowDate.getTime()
```

Comment:
Looks like this expression always will be `true`because any date will be lesser than now date

Reply:
No, because it is a condition for all calendar dates

Reply:
Understand, then could you please rename `lastCycleFinish`? Because now this variable looks like last cycle until today, but you mean that this cycle the end of the next expected cycle, which is calculated for the calendar

Reply:
Renamed